File: LayoutTests/http/tests/cache/resources/iframe304.php
<?php
require_once '../../resources/portabilityLayer.php';

clearstatcache();

if ($_SERVER["HTTP_IF_MODIFIED_SINCE"]) {
    header("HTTP/1.0 304 Not Modified");
    exit();
}
$one_year = 12 * 31 * 24 * 60 * 60;
$last_modified = gmdate(DATE_RFC1123, time() - $one_year);
$expires = gmdate(DATE_RFC1123, time() + $one_year);

header('Cache-Control: no-cache, max-age=' . $one_year);
header('Expires: ' . $expires);
header('Content-Type: text/html');
header('Content-Length: 0');
header('Etag: 123456789');
header('Last-Modified: ' . $last_modified);
exit();
?>

File: LayoutTests/http/tests/cache/resources/iframe304.py
#!/usr/bin/env python3

import os
import sys
from datetime import datetime, timedelta

modified_since = os.environ.get('HTTP_IF_MODIFIED_SINCE', '')

sys.stdout.write('Content-Type: text/html\r\n')

if modified_since:
    sys.stdout.write('status: 304\r\n\r\n')
    sys.exit(0)

one_year = 12 * 31 * 24 * 60 * 60
last_modified = '{} +0000'.format((datetime.utcnow() - timedelta(seconds=one_year)).strftime('%a, %d %b %Y %H:%M:%S'))
expires = '{} +0000'.format((datetime.utcnow() + timedelta(seconds=one_year)).strftime('%a, %d %b %Y %H:%M:%S'))

sys.stdout.write(
    f'Cache-Control: no-cache, max-age={one_year}\r\n'
    f'Expires: {expires}\r\n'
    'Content-Type: text/html\r\n'
    'Content-Length: 0\r\n'
    'Etag: 123456789\r\n'
    f'Last-Modified: {last_modified}\r\n\r\n'
)
sys.exit(0)